compute_tools/src/http/api.rs

```diff
@@ -165,6 +169,58 @@ async fn routes(req: Request<Body>, compute: &Arc<ComputeNode>) -> Response<Body
             }
         }
 
+        (&Method::POST, "/grants") => {
+            info!("serving /grants POST request");
+            let status = compute.get_status();
+            if status != ComputeStatus::Running {
+                let msg = format!(
+                    "invalid compute status for set_role_grants request: {:?}",
+                    status
+                );
+                error!(msg);
+                return Response::new(Body::from(msg));
+            }
+
+            let request = hyper::body::to_bytes(req.into_body()).await.unwrap();
+            let request = serde_json::from_slice::<SetRoleGrantsRequest>(&request).unwrap();
+
+            let privileges: Result<Vec<Privilege>, _> = request
+                .privileges
+                .iter()
+                .map(|p| Privilege::from_str(p.as_str()))
```

Comment:
I wanted some validation, is this too much effort? Could also just check that the string is in some array.